File: Sherlock-and-The-Beast.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'decentNumber' function below.
     *
     * The function accepts INTEGER n as parameter.
     */

    public static void decentNumber(int n)
    {
        for (int fives = n; fives >= 0; fives--)
        {
            if (fives % 3 == 0 && (n - fives) % 5 == 0)
            {
                Console.Write(new string('5', fives));
                Console.Write(new string('3', n - fives));
                Console.WriteLine();
                return;
            }
        }

        Console.WriteLine("-1");
    }
}


class Solution
{
    public static void Main(string[] args)
    {
        int t = Convert.ToInt32(Console.ReadLine().Trim());

        for (int tItr = 0; tItr < t; tItr++)
        {
            int n = Convert.ToInt32(Console.ReadLine().Trim());

            Result.decentNumber(n);
        }
    }
}

File: Sherlock-and-The-Beast.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'decentNumber' function below.
     *
     * The function accepts INTEGER n as parameter.
     */

    public static void decentNumber(int n) {
        for (int fives = n; fives >= 0; fives--) {

            if (fives % 3 == 0 && (n - fives) % 5 == 0) {

                for (int i = 0; i < fives; i++)
                    System.out.print("5");

                for (int i = 0; i < n - fives; i++)
                    System.out.print("3");

                System.out.println();
                return;
            }
        }

        System.out.println("-1");
    }
}

